### PR TITLE
Updated the changelog and version number

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:** updated the changelog to reflect (merged) PRs. Incremented the version number to `0.4.0-dev0` to reflect the development status of master.

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,79 @@
...
+  error messages for the user if incorrect syntax is used.
+  [#188](https://github.com/XanaduAI/pennylane/pull/188)
+
+* Adds support for multi-qubit expectation values of the `pennylane.Hermitian()`
+  observable [#192](https://github.com/XanaduAI/pennylane/pull/192)
+
+* Adds support for multi-qubit expectation values in `default.qubit`.
+  [#202](https://github.com/XanaduAI/pennylane/pull/202)
+
+* Organise templates into submodules [#195](https://github.com/XanaduAI/pennylane/pull/195).
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
* Organize templates into submodules [#195](https://github.com/XanaduAI/pennylane/pull/195).
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,79 @@
...
+  This included the following improvements:
+
+  - Distinguish embedding templates, layer templates and parameter templates
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
  - Distinguish embedding templates, layer templates, and parameter templates
```
:bowtie:

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,79 @@
...
+
+  - Add Random Circuit, in which rotations and 2-qubit gates are randomly
```
Comment (co9olguy) [inline .github/CHANGELOG.md]: Should we use the actual (code) template name?

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,79 @@
...
+
+* Fixes a bug in `Device.supported`, which would incorrectly
+  mark an operation as supported if it shared a name with an
+  observable [#203](https://github.com/XanaduAI/pennylane/pull/203)
+
+* Fixes a bug in `Operation.wires`, by explicitly casting the
+  type of each wire to an integer [#206](https://github.com/XanaduAI/pennylane/pull/206)
+
+* Removes code in PennyLane which configured the logger,
+  as this would clash with users configurations
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
  as this would clash with users' configurations
```

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,79 @@
...
+  [#211](https://github.com/XanaduAI/pennylane/pull/211)
+
+
+### Contributors
+
+This release contains contributions from:
+
+Shahnawaz Ahmed, riveSunder, Aroosa Ijaz, Josh Izaac, Nathan Killoran, Maria Schuld.
+
+Josh Izaac and Nathan Killoran.
```
Comment (co9olguy) [inline .github/CHANGELOG.md]: The contributors so nice they get listed twice?

File: .github/CHANGELOG.md (hunk 1 above)
Comment (josh146) [reply .github/CHANGELOG.md]: My :australia: was showing

File: .github/CHANGELOG.md (hunk 5 above)
Comment (josh146) [reply .github/CHANGELOG.md]: Oops, a copy paste error.
```suggestion
```

File: .github/CHANGELOG.md (hunk 3 above)
Comment (josh146) [reply .github/CHANGELOG.md]:
```suggestion
  - Added a random circuit template (`RandomLayers()`), in which rotations and 2-qubit gates are randomly
```

Comment (josh146) [reply .github/CHANGELOG.md]: done

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,78 @@
...
+* Adds support for multi-qubit expectation values of the `pennylane.Hermitian()`
+  observable [#192](https://github.com/XanaduAI/pennylane/pull/192)
+
+* Adds support for multi-qubit expectation values in `default.qubit`.
+  [#202](https://github.com/XanaduAI/pennylane/pull/202)
+
+* Organize templates into submodules [#195](https://github.com/XanaduAI/pennylane/pull/195).
+  This included the following improvements:
+
+  - Distinguish embedding templates, layer templates, and parameter templates
```
Comment (josh146) [inline .github/CHANGELOG.md]:
```suggestion
  - Distinguish embedding templates, layer templates, and parameter templates.
    Parameter templates are contained within the new submodule `pennylane.init`.
```

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -17,12 +17,13 @@
 * Adds support for multi-qubit expectation values in `default.qubit`.
   [#202](https://github.com/XanaduAI/pennylane/pull/202)
 
-* Organise templates into submodules [#195](https://github.com/XanaduAI/pennylane/pull/195).
+* Organize templates into submodules [#195](https://github.com/XanaduAI/pennylane/pull/195).
   This included the following improvements:
 
-  - Distinguish embedding templates, layer templates and parameter templates
+  - Distinguish embedding templates, layer templates, and parameter templates.
+    Parameter templates are contained within the new submodule `pennylane.init`.
```
Comment (co9olguy) [inline .github/CHANGELOG.md]: Are we calling them `parameter templates`? Probably better to just say we added new random initialization functions supporting the templates

Comment (josh146) [reply .github/CHANGELOG.md]: That sounds a lot better, I've made the suggested update.

**Pull request merged**